src/tests/flask/security_route_middleware_integration_test.py

```diff
@@ -0,0 +1,108 @@
+import jwt
+from injector import Injector
+
+from src.constants.http import HttpMethods
+from src.constants.oauth2 import Roles, Scopes
+from src.core.di_config import DiProviders
+from src.env_config import get_env_config_instance
+from src.flask.routes.security_route_middleware import setup_security_route_middleware
+from src.tests.flask.before_request_middleware_function import before_request_middleware
+from src.tests.utils.fixtures.endpoint_request_fixture import create_request
+from src.tests.utils.fixtures.token_fixture import (
+    build_user_access_token,
+    build_expire_access_token,
+    build_invalid_access_token,
+)
+from src.utils import oauth2_util
+
+
+class SecurityRouteMiddlewareIntegrationTest(unittest.TestCase):
+    def setUp(self):
+        self.env_config = get_env_config_instance()
+
+        self.ioc = Injector(DiProviders)
+        self.security_route_middleware_blueprint = setup_security_route_middleware(
+            self.ioc
+        )
+
+    @mock.patch(
+        "src.tests.flask.before_request_middleware_function.oauth2_util.validate_roles_and_scopes",
+        return_value=oauth2_util.validate_roles_and_scopes,
+    )
+    @mock.patch(
+        "src.tests.flask.before_request_middleware_function.oauth2_util.is_endpoint_protected",
+        return_value=oauth2_util.is_endpoint_protected,
+    )
+    def test_before_request_protected_endpoints(
+        self, mocked_func_is_endpoint_protected, mocked_func_validate_roles_and_scopes
+    ):
+        access_token = build_user_access_token(
+            roles=Roles.ADMINISTRATOR.value, scopes=Scopes.READ.value
+        )
+        request = create_request(
+            url="/chefs/<chef_id>",
+            method=HttpMethods.GET.value,
+            access_token=access_token,
+        )
+
+        self.security_route_middleware_blueprint.before_app_request(
+            before_request_middleware(self.ioc, request)
+        )
+        mocked_func_is_endpoint_protected.assert_called_with(request)
+        mocked_func_validate_roles_and_scopes.assert_called_with(
+            self.env_config.oauth2_secret_key, request
+        )
+
+    @mock.patch(
+        "src.tests.flask.before_request_middleware_function.make_response",
+        return_value=jwt.exceptions.InvalidSignatureError,
+    )
+    def test_before_request_invalid_access_token(self, mocked_invalid_token):
+        invalid_access_token = build_invalid_access_token()
+        request = create_request(
+            url="/chefs/<chef_id>",
+            method=HttpMethods.GET.value,
+            access_token=invalid_access_token,
+        )
+
+        response = self.security_route_middleware_blueprint.before_app_request(
+            before_request_middleware(self.ioc, request)
+        )
+        self.assertEqual(response, jwt.exceptions.InvalidSignatureError)
+
+    @mock.patch(
+        "src.tests.flask.before_request_middleware_function.make_response",
+        return_value=jwt.exceptions.ExpiredSignatureError,
```

Comment:
This is not correct, you should be able to raise an error and not "return an error", the same comment for all the other similar cases

Reply:
I made it that way mocking the `make_response`. The Error is raising, the thing is that when the error rise it return a `make_responde` but this doesn't work out of flask app context, as i'm running the function out of flask i cant generate responses.

Reply:
I see, this is fine I think